integration-cli/docker_cli_save_load_test.go

```diff
@@ -97,6 +97,56 @@ func (s *DockerSuite) TestSaveSingleTag(c *check.C) {
 	}
 }
 
+// TestSaveAndLoadMultipleTags test fix for Issue #10592.
+func (s *DockerSuite) TestSaveAndLoadMultipleTags(c *check.C) {
+	repoNameOne := "save-load-multiple-tags-one"
+	repoNameTwo := "save-load-multiple-tags-two"
+
+	// Tag emptyfs:latest with two more tags.
+	dockerCmd(c, "tag", "emptyfs:latest", fmt.Sprintf("%v:latest", repoNameOne))
+	dockerCmd(c, "tag", "emptyfs:latest", fmt.Sprintf("%v:latest", repoNameTwo))
+
+	tmpDir, err := ioutil.TempDir("", "save-load-multiple-tags")
+	if err != nil {
+		c.Errorf("failed to create temporary directory: %s", err)
+	}
+
+	out, _ := dockerCmd(c, "images", "-q", repoNameOne)
+	cleanedShortImageID := strings.TrimSpace(out)
+
+	// Save the image into a tarball. This should preserve all three tags.
+	saveCmd := exec.Command(dockerBinary, "save", cleanedShortImageID)
+	saveOut, err := saveCmd.CombinedOutput()
+	if err != nil {
+		c.Errorf("failed to run docker save command: %s", err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(tmpDir, "emptyfs.tar"), saveOut, 0666); err != nil {
+		c.Errorf("failed to create tar ball: %s", err)
+	}
+
+	// Untag tags: save-load-multiple-tags-one and save-load-multiple-tags-two.
+	rmiCmd := exec.Command(dockerBinary, "rmi", repoNameOne, repoNameTwo)
+	if _, err := rmiCmd.CombinedOutput(); err != nil {
+		c.Errorf("failed to run docker rmi command: %s", err)
+	}
+
+	// This load should populate the non-conflicting tags back.
+	// After this load, the image should have all 3 tags.
+	loadCmd := exec.Command(dockerBinary, "load", "-i", filepath.Join(tmpDir, "emptyfs.tar"))
+	if _, err := loadCmd.CombinedOutput(); err != nil {
+		c.Errorf("failed to run docker load command: %s", err)
+	}
+
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "images"),
+		exec.Command("grep", "-E", "(save-load-multiple-tags-one|save-load-multiple-tags-two)"),
```

Comment:
I'm not sure this pipeline is very portable. I think you can just call docker images and check if both tags are in the output, can't you?